### ISSUE TITLE
<Carousel/> - clickNext doesn't call beforeChange and afterChange callbacks

### ISSUE THREAD

**Issue by davydof** [wix/wix-style-react#6336]
# 🐛 Bug Report

### 🏗 Relevant Components

Carousel

### 😯 Current Behavior

```js
await CarouselTestkit({...}).clickNext()
await CarouselTestkit({...}).clickNext()
```

`afterChange` is not triggered at all
`beforeChange` is triggered only once after first `clickNext` and didn't trigger after second `clickNext`

### 🤔 Expected Behavior

```js
await CarouselTestkit({...}).clickNext()
await CarouselTestkit({...}).clickNext()
```

`afterChange` is triggered two times
`beforeChange` is triggered two times

### 👣 Steps to Reproduce

```js
<Carousel
  dataHook="some-carousel"
  afterChange={(after) => {
    console.log("afterChange after", after);
  }}
  beforeChange={(before, after) => {
    console.log("beforeChange before, after", before, after);
  }}
  images={[
    {
      src: "1.jpg",
    },
    {
      src: "2.jpg",
    },
    {
      src: "3.jpg",
    },
    {
      src: "4.jpg",
    },
  ]}
/>;
```
  - **WSR version**: `9.44.0`
  - **Yoshi version**: `4.253.0`

### 👀 Severity
Critical

**Comment by davydof** [comment]
It works like:
```js
jest.useFakeTimers();
await driver.when.nextImageButtonClicked();
jest.runAllTimers();
await driver.when.nextImageButtonClicked();
jest.useRealTimers();
```

Should I handle this in my driver or probably it can be implemented inside Carousel testkit?

**Comment by EgoziE** [comment]
Moved to Jira - closing
